src/main/perl/Check.pm

```diff
@@ -536,28 +536,28 @@ sub link ($$;%) {
     $target = _realpath($otarget);
     $message = $opt{hard} ? "create hard link $osource => $otarget" :
 	"create symlink $osource -> $otarget";
-    #
-    # check target
-    #
-    if ($otarget =~ /^\//) {
-	$test = $target;
-    } else {
-	$test = $source;
-	$test =~ s=[^/]+$==;
-	$test .= $otarget;
-    }
+
     if ($opt{hard}) {
-	# note: we follow symlinks for $target to be compatible with ln(1)
-	@stat = LC::Fatal::stat($test);
+        # Check target that is required to exist.
+	@stat = LC::Fatal::lstat($target);
 	if (@stat) {
-	    if (-d _) {
+	    if (-d _ && ! -l _) {
```

Comment:
I was about to look at what was the `_` magic doing! In this case this is even better.